docs/src/piccolo/query_clauses/for_update.rst

```diff
@@ -0,0 +1,62 @@
+.. _limit:
+
+for_update
+=====
+
+You can use ``for_update`` clauses with the following queries:
+
+* :ref:`Objects`
+* :ref:`Select`
+
+Returns a query that will lock rows until the end of the transaction, generating a SELECT ... FOR UPDATE SQL statement.
+
+.. note:: Postgres and CockroachDB only.
+
+-------------------------------------------------------------------------------
+
+default
+~~~~~~~
+To use select for update without extra parameters. All matched rows will be locked until the end of transaction.
+
+.. code-block:: python
+
+    await Band.select(Band.name == 'Pythonistas').for_update()
+
+
+equals to:
+
+.. code-block:: sql
+
+   SELECT ... FOR UPDATE
+
+
+nowait
+~~~~~~~
+If another transaction has already acquired a lock on one or more selected rows, the exception will be raised instead of waiting for another transaction
+
+
+.. code-block:: python
+
+    await Band.select(Band.name == 'Pythonistas').for_update(nowait=True)
+
+
+skip_locked
+~~~~~~~
+Ignore locked rows
+
+.. code-block:: python
+
+    await Band.select(Band.name == 'Pythonistas').for_update(skip_locked=True)
+
+
+
+of
+~~~~~~~
```

Comment:
Can we add a blank line under the headings, and make the underlines match the width of the title? It's a minor thing that RST linters complain about.